### PR TITLE
feat(CMF): add registry lock + warning if already registred

### PR DESCRIPTION
**Pull request by jsomsanith**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
When user want to register
* an item in a a locked registry, it just doesn't add it, silently. It might lead to error when the app wants to access to this not-so-registered item.
* an item with an existing key, it just override it silently. But that might be a usage error.

**What is the new behavior?**
With a locked registry, every add will result in a error.
With an existing key, any override attempt will print a warning that their configuration might be wrong.

Expose `lock()` function.

**Does this PR introduce a breaking change?**

- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...
Any attempt to register in a locked registry will throw an error.

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://165.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://165.talend.surge.sh)

File: packages/cmf/src/registry.js [hunk 1]
```diff
@@ -33,11 +33,22 @@ const Registry = {
...
+	if (Registry.isLocked()) {
+		throw new Error(
+			`CMF: The registry is locked, you cannot therefore add '${id}' in it. ` +
+			'Please check your CMF configuration, it should not move after the initial ' +
+			'configuration before bootstrap.'
+		);
 	}
-	// Should it do nothing ?
+
+	const reg = Registry.getRegistry();
```
Comment (talend-glorieux) [inline packages/cmf/src/registry.js]: Why not call it registry?

File: packages/cmf/src/registry.js [hunk 2]
```diff
@@ -33,11 +33,22 @@ const Registry = {
...
+	if (reg[id]) {
+		console.warn( // eslint-disable-line no-console
+			`CMF: The '${id}' object is registered, overriding and existing '${id}' object. ` +
```
Comment (talend-glorieux) [inline packages/cmf/src/registry.js]: "overriding an existing"?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://165.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://165.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://165.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://165.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://165.talend.surge.sh)

**Pull request merged**